### PR TITLE
x86: Fix access flags for SHR/SHL/SAL/SAR

### PR DESCRIPTION
**Pull request by fvrmatteo**
The changes must be checked using the Intel Instructions Manual.

I noticed uploading using the web interface of Github added a CR/LF on both files the first time, sorry.

### CODE REVIEW

File: arch/X86/X86MappingInsnOp.inc [hunk 1]
```diff
@@ -9811,19 +9811,19 @@
...
 	X86_EFLAGS_MODIFY_OF | X86_EFLAGS_MODIFY_SF | X86_EFLAGS_MODIFY_ZF | X86_EFLAGS_UNDEFINED_AF | X86_EFLAGS_MODIFY_PF | X86_EFLAGS_MODIFY_CF,
-	{ CS_AC_READ, CS_AC_IGNORE, 0 }
+	{ CS_AC_READ | CS_AC_WRITE, CS_AC_IGNORE, 0 }
 },
 {	/* X86_SHR8mCL, X86_INS_SHR: shr{b}	$dst, cl */
 	X86_EFLAGS_MODIFY_OF | X86_EFLAGS_MODIFY_SF | X86_EFLAGS_MODIFY_ZF | X86_EFLAGS_UNDEFINED_AF | X86_EFLAGS_MODIFY_PF | X86_EFLAGS_MODIFY_CF,
-	{ CS_AC_READ, CS_AC_READ, 0 }
+	{ CS_AC_READ | CS_AC_WRITE, CS_AC_READ, 0 }
 },
 {	/* X86_SHR8mi, X86_INS_SHR: shr{b}	$dst, $src */
```
Comment (aquynh) [inline arch/X86/X86MappingInsnOp.inc]: note that the instructions (such as X86_SHR8mi) has immediate operand, indicated by "i" in its name. and immediate operand should have CS_AC_IGNORE access. therefore, some of your changes are incorrect.

can you please fix them?

**Comment by fvrmatteo** [comment]
I'm really noob at using git, uff >.>

Anyway, you are right, I misinterpreted the meaning of "i". If it is an immediate, I should have fixed it now.

**Comment by aquynh** [comment]
no worry. next time you dont need to open a new pull request, but just need to push your changes to the same PR branch, they will show up here. that is it, now you already know everything you need to know about using Github :-)

File: arch/X86/X86MappingInsnOp_reduce.inc [hunk 2]
```diff
@@ -5267,11 +5267,11 @@
 },
 {	/* X86_SHRD16mrCL, X86_INS_SHRD: shrd{w}	$dst, $src2, cl */
 	X86_EFLAGS_UNDEFINED_OF | X86_EFLAGS_MODIFY_SF | X86_EFLAGS_MODIFY_ZF | X86_EFLAGS_UNDEFINED_AF | X86_EFLAGS_MODIFY_PF | X86_EFLAGS_MODIFY_CF,
-	{ CS_AC_READ, CS_AC_READ, CS_AC_READ, 0 }
+	{ CS_AC_READ | CS_AC_WRITE, CS_AC_READ, CS_AC_READ, 0 }
 },
 {	/* X86_SHRD16mri8, X86_INS_SHRD: shrd{w}	$dst, $src2, $src3 */
 	X86_EFLAGS_UNDEFINED_OF | X86_EFLAGS_MODIFY_SF | X86_EFLAGS_MODIFY_ZF | X86_EFLAGS_UNDEFINED_AF | X86_EFLAGS_MODIFY_PF | X86_EFLAGS_MODIFY_CF,
-	{ CS_AC_READ, CS_AC_READ, CS_AC_IGNORE, 0 }
+	{ CS_AC_READ | CS_AC_WRITE, CS_AC_READ, CS_AC_READ, 0 }
```
Comment (aquynh) [inline arch/X86/X86MappingInsnOp_reduce.inc]: you forgot to fix this CS_AC_READ?

File: arch/X86/X86MappingInsnOp_reduce.inc [hunk 3]
```diff
@@ -4691,7 +4691,7 @@
 },
 {	/* X86_SAR8ri, X86_INS_SAR: sar{b}	$dst, $src2 */
 	X86_EFLAGS_MODIFY_OF | X86_EFLAGS_MODIFY_SF | X86_EFLAGS_MODIFY_ZF | X86_EFLAGS_UNDEFINED_AF | X86_EFLAGS_MODIFY_PF | X86_EFLAGS_MODIFY_CF,
-	{ CS_AC_READ | CS_AC_WRITE, CS_AC_IGNORE, 0 }
+	{ CS_AC_READ | CS_AC_WRITE , CS_AC_IGNORE, 0 }
```
Comment (aquynh) [inline arch/X86/X86MappingInsnOp_reduce.inc]: please remove this extra space

File: arch/X86/X86MappingInsnOp_reduce.inc (hunk 2 above)
Comment (fvrmatteo) [inline arch/X86/X86MappingInsnOp_reduce.inc]: Ok, I'm doing something wrong. When you write: "mri8" what does it means?
I interpreted it as:
- $dst is "m = memory"
- $src2 is "r = register"
- $src3 is "i8 = immediate 8 bit"

Comment (aquynh) [inline arch/X86/X86MappingInsnOp_reduce.inc]: yes, you are right. that is why CS_AC_READ must be changed it back to CS_AC_IGNORE

**Comment by fvrmatteo** [comment]
Ok, updated again, sorry for the mess :D

**Comment by aquynh** [comment]
merged, thanks!

**Pull request merged**